Sources/Sentry/SentryHttpTransport.m

```diff
@@ -36,35 +36,15 @@ - (id)initWithOptions:(SentryOptions *)options
         self.envelopeRateLimit = envelopeRateLimit;
 
         [self setupQueueing];
-        [self sendCachedEventsAndEnvelopes];
+        [self sendAllCachedEnvelopes];
     }
     return self;
 }
 
-// TODO: needs refactoring
 - (void)sendEvent:(SentryEvent *)event
```

Comment:
H: What about we get rid of this method too?
Leave the responsability of wraping stuff in envelopes to the caller?

Reply:
As far as I understand envelopes they are an internal format of us to send SDK data over HTTP to Sentry. This is very specific to the transport and not the client. The client shouldn't care and know how events are sent. If we remove this here the client would know. I would even argue that we should remove the `sendEnvelope` instead and add methods for sending sessions and such. Only the transport and his interacting classes should know what an envelope is. If at some point we decide that we introduce another format of sending our data to sentry we only need to touch the transport and not the client and the hub. For me, it is a good sign, that we don't have to change a single line of code in the client in this PR.

Reply:
IMO `send(final @NotNull SentryEnvelope envelope)` makes more sense as the transport could send a bunch of items at once in the same envelope.
if we don't expose a method like this, it'd be hard to send for example, an event (error) and a session update.
`SentryClient` could have methods specialized in its type like captureEvent and so on, but all of them forwards to `transport.sendEnvelope`

Reply:
Hm, I get your point. We get more flexibility when the transport only has `sendEnvelope`. Removing it is not an option. The only thing I don't like about this is that the client has the responsibility of building the events and also translating them to envelopes. Currently, this would make testing a huge pain because we currently verify what the transport receives. 
I think we are missing a component inside the client that maps everything to envelopes so we separate the responsibility and make testing easier. The advantage with this is we have maximum flexibility with the transport and somebody implementing a custom transport doesn't have to worry about mapping everything to envelopes. What do you think about this @marandaneto?

Reply:
I had a discussion with @HazAT today and he is in favor of keeping the `sendEvent` as he also did it like this in sentry-javascript, because of the reasons I mentioned above. I think we still can keep the `sendEnvelope` to offer more flexibility, but the transport should take care of creating the proper envelopes for events, sessions, transactions, ...

Reply:
Given that the transport isn't public, it's not a big issue. Also we're using sendEvent a lot for testing so lots of work to get rid of it.
We agree that going forward we'd like to get rid of it though.